### PR TITLE
fix: add warnings for unsupported feature flags (#18672)

### LINKED ISSUE #18672: Log or throw for non-existent feature flags
### Describe your motivation

When adding a non-existent feature flag to `vaadin-featureflags.properties`, the app starts normally without errors or any indication in the logs that something unexpected happened. This behavior is problematic (a) for people upgrading to newer versions where the feature flag is no longer working, and (b) when typos are made in the feature flag name. 

### Describe the solution you'd like

During startup, an error should be thrown, or a warning should be logged, if a non-supported feature flag is present in  `vaadin-featureflags.properties`.

### PR DESCRIPTION
**Pull request by schastlivcev**
## Description

There was no indication if you were using non-existent or unsupported feature flags at startup. This change brings log warnings about existence of such feature flags

Fixes #18672

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

**Comment by schastlivcev** [comment]
Hi, I'm sorry for taking so long, have some questions:
1. I saw "expressBuild" feature flag in some .properties files, should I add it to the FeatureFlags class? #15375
2. I checked against the ids in the class, is it correct or is there another place to look?
3. Original issue was about the .properties file, should I expand it to the system properties?
4. Do I need to put a boolean check flag into a synchronized block?

**Comment by mshabarov** [comment]
Thanks for your contribution @schastlivcev !

Some suggestions:
1. ExpressBuild flag has been removed and isn't used anymore. It was related to [pre-compiled bundles](https://vaadin.com/docs/latest/configuration/development-mode#precompiled-bundle) feature, but can be removed now from everywhere in Flow.
2. I think the `com.vaadin.experimental.` + flag's id is enough for matching.
3. Would be better to apply to system properties, as this might be a common case to set flags in CI for instance.

File: flow-server/src/main/java/com/vaadin/experimental/FeatureFlags.java [hunk 1]
```diff
@@ -339,11 +355,39 @@ private boolean isDevelopmentMode() {
...
+            checkForUnsupportedFeatureFlags(filteredSystemProps, this::getSystemPropertyName);
+            isSystemPropertiesChecked = true;
+        }
+    }
+
+    private void checkForUnsupportedFeatureFlags(Properties props, Function<String, String> propertyWithPrefix) {
+        for (Object property : props.keySet()) {
+            if (features.stream()
+                    .noneMatch(feature -> propertyWithPrefix.apply(feature.getId()).equals(property))) {
+                getLogger().warn("Unsupported feature flag is present: {}", property);
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/experimental/FeatureFlags.java:389]: I suggest to log also whether system property or property file has unsupported flag.

Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/experimental/FeatureFlags.java:389]: Let me add this later in a separate PR.

**Review by mshabarov — APPROVED** [review]

**Comment by mshabarov** [comment]
@schastlivcev anything else you wanted to add? Otherwise, I'll merge this pull request.
Thanks again for contribution!

**Comment by mshabarov** [comment]
@schastlivcev could you please run `mvn formatter:format` and push changes? formatting blocks this PR and I cannot access your fork branch.

**Comment by schastlivcev** [comment]
Sorry for so long replies, lots of work. I'll deal with changes by the end of the week

**Comment by mshabarov** [comment]
Closed in favour of https://github.com/vaadin/flow/pull/19345.

**Pull request closed without merging**